### PR TITLE
GCI-776 show order total cost when checking out

### PR DESCRIPTION
**Pull request by rjadhav-ch**
Implements logic to calculate `total_order_cost` of `checkout` object.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java [hunk 1]
```diff
@@ -46,10 +43,30 @@ public Checkout createCheckout(Item item, String userId, String email, DeliveryD
         checkout.getData().setReference(objectId);
         checkout.getData().setKind("order");
         checkout.getData().setDeliveryDetails(deliveryDetails);
+        String totalOrderCostStr = calculateTotalOrderCostForCheckout(checkout) + "";
+        checkout.getData().setTotalOrderCost(totalOrderCostStr);
 
         return checkoutRepository.save(checkout);
     }
 
+    protected double calculateTotalOrderCostForCheckout(Checkout checkout){
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java:52]: I note we used integers for previous cost calculations.

File: src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java [hunk 2]
```diff
@@ -46,10 +43,30 @@ public Checkout createCheckout(Item item, String userId, String email, DeliveryD
         checkout.getData().setReference(objectId);
         checkout.getData().setKind("order");
         checkout.getData().setDeliveryDetails(deliveryDetails);
+        String totalOrderCostStr = calculateTotalOrderCostForCheckout(checkout) + "";
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java:46]: The `CheckoutService` could delegate the calculation to an instance of a separate class. The calculation is not complex, but having it the responsibility of a separate class/component/service might also lend itself to tests of any calculation corner cases there may be?

File: src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java (hunk 1 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java]: updated

File: src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java (hunk 2 above)
Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/orders/api/service/CheckoutService.java]: Calculation logic moved to a helper class.

File: src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Test
+    @DisplayName("calculateTotalOrderCostForCheckout sums `total order cost` correctly")
+    void calculateTotalOrderCostForCheckout(){
+        // Given
+        Checkout checkout = new Checkout();
+        checkout.setData(createCheckoutData());
+
+        // When
+        double actualTotalOrderCost = CheckoutHelper.calculateTotalOrderCostForCheckout(checkout);
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java:41]: `int`?

File: src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Test
+    @DisplayName("calculateTotalOrderCostForCheckout returns zero `total order cost` when no itemcosts in item")
+    void calculateTotalOrderCostForCheckoutNoItemCosts(){
+        // Given
+        Checkout checkout = new Checkout();
+        checkout.setData(new CheckoutData());
+
+        // When
+        double actualTotalOrderCost = CheckoutHelper.calculateTotalOrderCostForCheckout(checkout);
```
Comment (VaughanJackson123) [inline src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java:55]: `int`?

File: src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java (hunk 3 above)
Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java]: fixed

File: src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java (hunk 4 above)
Comment (rjadhav-ch) [reply src/test/java/uk/gov/companieshouse/orders/api/util/CheckoutHelperTest.java]: fixed

File: src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.orders.api.model.Checkout;
+import uk.gov.companieshouse.orders.api.model.CheckoutData;
+import uk.gov.companieshouse.orders.api.model.Item;
+import uk.gov.companieshouse.orders.api.model.ItemCosts;
+
+import java.util.List;
+
+@Component
+public class CheckoutHelper {
```
Comment (VaughanJackson123) [inline src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java:12]: This could be seen as a calculator - its only responsibility is calculating the total cost.

Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java:12]: Which is why it was originally a static method. :)

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java:12]: I am guessing it was static because the class has no state of its own.

Comment (rjadhav-ch) [reply src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java:12]: yes. I'll revert it to a static method.

Comment (VaughanJackson123) [reply src/main/java/uk/gov/companieshouse/orders/api/util/CheckoutHelper.java:12]: It's not a big deal, but I think the flexibility that OO and DI give you apply even when it happens that the class/function has no state. :)
